src/Makefile.am

```diff
@@ -33,100 +33,171 @@ d_bindir = $(top_builddir)/bin
+			    trampolines.cpp
+
+libnohijack_a_SOURCES = dmtcpnohijackstubs.cpp			\
+			nosyscallsreal.c
+
+__d_libdir__libdmtcp_so_SOURCES = alarm.cpp			\
+				  ckptserializer.cpp 		\
+				  dmtcpplugin.cpp 		\
+				  dmtcpworker.cpp 		\
+				  execwrappers.cpp 		\
+				  glibcsystem.cpp 		\
+				  miscwrappers.cpp 		\
+				  plugininfo.cpp 		\
+				  pluginmanager.cpp		\
+				  popen.cpp 			\
+				  signalwrappers.cpp 		\
+				  siginfo.cpp 			\
+				  syslogwrappers.cpp 		\
+				  terminal.cpp 			\
+				  threadlist.cpp 		\
+				  threadsync.cpp 		\
+				  threadwrappers.cpp 		\
+				  writeckpt.cpp
+
+__d_libdir__libdmtcp_so_LDFLAGS = -shared -Xlinker -znow
+
+# Note that an ELF object uses libsyscallsreal.a or libnohijack.a
+#  but not both.  libnohijack.a has stub definitions for same symbols.
+__d_libdir__libdmtcp_so_LDADD = libdmtcpinternal.a 		\
+				 libjalib.a 			\
+				 libsyscallsreal.a 		\
+				 mtcp/libmtcp.a			\
+				 -ldl -lpthread -lrt
 
-__d_bindir__dmtcp_coordinator_SOURCES = dmtcp_coordinator.cpp lookup_service.cpp restartscript.cpp
 
 __d_bindir__dmtcp_nocheckpoint_SOURCES = dmtcp_nocheckpoint.c
 
-__d_bindir__dmtcp_restart_SOURCES = dmtcp_restart.cpp util_exec.cpp
+__d_bindir__dmtcp_coordinator_SOURCES = dmtcp_coordinator.cpp 	\
+					lookup_service.cpp 	\
+					restartscript.cpp
 
-__d_bindir__dmtcp_command_SOURCES = dmtcp_command.cpp
+__d_bindir__dmtcp_coordinator_LDADD = libdmtcpinternal.a 	\
+				      libjalib.a 		\
+				      libnohijack.a		\
+				      -lpthread -lrt
 
-__d_libdir__libdmtcp_so_SOURCES = dmtcpworker.cpp threadsync.cpp \
-		      coordinatorapi.cpp execwrappers.cpp \
-		      signalwrappers.cpp \
-		      terminal.cpp \
-		      alarm.cpp \
-		      threadwrappers.cpp \
-		      miscwrappers.cpp ckptserializer.cpp writeckpt.cpp \
-		      glibcsystem.cpp \
-		      threadlist.cpp siginfo.cpp \
-		      dmtcpplugin.cpp popen.cpp syslogwrappers.cpp \
-		      dmtcp_dlsym.cpp plugininfo.cpp pluginmanager.cpp
+__d_bindir__dmtcp_launch_SOURCES = dmtcp_launch.cpp
 
-__d_libdir__libdmtcp_so_LDFLAGS = -shared -Xlinker -znow
+__d_bindir__dmtcp_launch_LDADD  = libdmtcpinternal.a 		\
+				  libjalib.a			\
+				  libnohijack.a			\
+				  -lpthread -lrt -ldl
 
-# Note that an ELF object uses libsyscallsreal.a or libnohijack.a
-#  but not both.  libnohijack.a has stub definitions for same symbols.
-__d_libdir__libdmtcp_so_LDADD = libdmtcpinternal.a libjalib.a \
-				libsyscallsreal.a mtcp/libmtcp.a \
-				-ldl -lpthread -lrt
-
-__d_bindir__dmtcp_coordinator_LDADD = libdmtcpinternal.a libjalib.a \
-			  libnohijack.a -lpthread -lrt
-__d_bindir__dmtcp_launch_LDADD  = libdmtcpinternal.a libjalib.a \
-			  libnohijack.a -lpthread -lrt -ldl
-__d_bindir__dmtcp_restart_LDADD     = libdmtcpinternal.a libjalib.a \
-			  libnohijack.a -lpthread -lrt -ldl
-__d_bindir__dmtcp_command_LDADD     = libdmtcpinternal.a libjalib.a \
-			  libnohijack.a -lpthread -lrt -ldl
+__d_bindir__dmtcp_restart_SOURCES = dmtcp_restart.cpp
```

Comment:
Is  `util_exec.cpp` required here?

Reply:
It's part of libdmtcpinternal.a, which is part of LDADD and hence isn't required here explicitly.